.github/workflows/ci.yml

```diff
@@ -0,0 +1,171 @@
+  PACKAGE_PROJECT_PATH: '${{ github.workspace }}\src\Files.App (Package)\Files.Package.wapproj'
+  INTERACTION_TESTS_PROJECT_DIR: '${{ github.workspace }}\tests\Files.InteractionTests'
+  INTERACTION_TESTS_PROJECT_PATH: '${{ github.workspace }}\tests\Files.InteractionTests\Files.InteractionTests.csproj'
+  WORKING_DIR: ${{ github.workspace }} # Default: 'D:\a\Files\Files'
+  ARTIFACTS_STAGING_DIR:  ${{ github.workspace }}\artifacts
+  APPX_PACKAGE_DIR: ${{ github.workspace }}\artifacts\AppxPackages
+  APPX_SELFSIGNED_CERT_PATH: '${{ github.workspace }}\builds\Files_SelfSigned.pfx'
+
+jobs:
+  build:
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        configuration: [Debug]
+        platform: [x64, arm64]
+    env:
+      CONFIGURATION: ${{ matrix.configuration }}
+      ARCHITECTURE: ${{ matrix.platform }}
+
+    steps:
+    - name: Checkout the repository
+      uses: actions/checkout@v3
+
+    - name: Setup MSBuild
+      uses: microsoft/setup-msbuild@v1
+
+    - name: Setup NuGet
+      uses: NuGet/setup-nuget@v1.1.1
+
+    - name: Setup .NET 7
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: '7.0.x'
+
+    - name: Restore NuGet
+      shell: pwsh
+      run: 'nuget restore $env:SOLUTION_PATH'
+
+    - name: Restore Files
+      shell: pwsh
+      run: |
+        msbuild $env:SOLUTION_PATH `
+        -t:Restore `
+        -p:Platform=$env:ARCHITECTURE `
+        -p:Configuration=$env:CONFIGURATION `
+        -p:PublishReadyToRun=true
+
+    - if: env.ARCHITECTURE != env.PRIMARY_ARCHITECTURE
+      name: Build Files
+      run: |
+        msbuild $env:PACKAGE_PROJECT_PATH `
+        -t:Build `
+        -p:Configuration=$env:CONFIGURATION `
+        -p:Platform=$env:ARCHITECTURE `
+        -p:AppxBundle=Never
+  
+    - if: env.ARCHITECTURE == env.PRIMARY_ARCHITECTURE
+      name: Build & Publish Files
+      run: |
+        msbuild $env:PACKAGE_PROJECT_PATH `
+        -t:Build `
+        -t:_GenerateAppxPackage `
+        -p:Configuration=$env:CONFIGURATION `
+        -p:Platform=$env:ARCHITECTURE `
+        -p:AppxBundlePlatforms=$env:PRIMARY_ARCHITECTURE `
+        -p:AppxBundle=Always `
+        -p:UapAppxPackageBuildMode=SideloadOnly `
+        -p:AppxPackageDir=$env:APPX_PACKAGE_DIR `
+        -p:AppxPackageSigningEnabled=true `
+        -p:PackageCertificateKeyFile=$env:APPX_SELFSIGNED_CERT_PATH `
+        -p:PackageCertificatePassword="" `
+        -p:PackageCertificateThumbprint=""
+
+    - if: env.ARCHITECTURE == env.PRIMARY_ARCHITECTURE
+      name: Upload the packages to GitHub Actions
+      uses: actions/upload-artifact@v3
+      with:
+        name: 'Appx Packages (${{ env.CONFIGURATION }}, ${{ env.PRIMARY_ARCHITECTURE }})'
+        path: ${{ env.ARTIFACTS_STAGING_DIR }}
+
+  test:
+    needs: [build]
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        configuration: [Debug]
+        platform: [x64]
```

Comment:
Can we do this for the ARM version as well?

Reply:
Sadly, GitHub-hosted runner doesn's support Windows ARM64 environment, however self-hosted runner supports it as a public beta.

Reply:
definitely not a blocker for now but might be nice to do this in the future.